lunabot_control/scripts/differential_drive_controller.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import numpy as np
+import rospy
+from geometry_msgs.msg import Twist
+
+from lunabot_msgs.msg import RobotEffort, RobotState
+
+
+def ang_delta(deg, prev_deg):
+    raw = deg - prev_deg
+    if raw == 0:
+        return 0
+    turn = min(raw % 360.0, -raw % 360.0)
+    dir_ = abs(raw) / raw if turn == raw else -abs(raw) / raw
+    return turn * dir_
```

Comment:
Might run into problems if deg is more than 360 over prev_deg. The following might be better:

raw = deg - prev_deg
raw %= 360
if(raw < 0)
  raw += 360
raw -= 180 (to bound between -180 to 180)

Reply:
python mod doesn't output negative numbers, the if statement branch would never be run

Reply:
you can remove the if statement, you can just do raw %= 360 then raw -= 180

Reply:
given that this function will be ran very fast > 100hz, it may be a fair assumption that deg may not be more than 360 over prev_deg. it has also been moved to the teensy_driver_node to avoid doing velocity calculation here